pandas/core/indexes/datetimelike.py

```diff
@@ -445,17 +445,11 @@ def isin(self, values):
 
         return algorithms.isin(self.asi8, values.asi8)
 
+    @Appender(_index_shared_docs['repeat'] % _index_doc_kwargs)
     def repeat(self, repeats, *args, **kwargs):
-        """
-        Analogous to ndarray.repeat.
-        """
         nv.validate_repeat(args, kwargs)
-        if is_period_dtype(self):
-            freq = self.freq
-        else:
-            freq = None
-        return self._shallow_copy(self.asi8.repeat(repeats),
-                                  freq=freq)
+        freq = self.freq if is_period_dtype(self) else None
+        return self._shallow_copy(self.asi8.repeat(repeats), freq=freq)
```

Comment:
I've deleted the `PeriodIndex` implementation in favor of this, but there looks to be a bug in `PeriodIndex` that causes this to break for frequencies with a multiplier, e.g. `'2D'` (xref #24391).  I've xfailed the associated broken test, but can reinstate the `PeriodIndex` implementation if we'd rather have a fully working version.